core/keyboard_nav/key_map.js

```diff
@@ -99,18 +93,17 @@ Blockly.user.keyMap.getActionByKeyCode = function(keyCode) {
 };
 
 /**
- * Get the serialized key that corresponds to the action name.
- * @param {!string} actionName The name of the action.
+ * Get the serialized key that corresponds to the action.
+ * @param {!Blockly.Action} action The action for which we want to get
+ *     the key.
  * @return {string} The serialized key or null if the action does not have
  *     a key mapping.
  * @package
  */
-Blockly.user.keyMap.getKeyByActionName = function(actionName) {
+Blockly.user.keyMap.getKeyByAction = function(action) {
   var keys = Object.keys(Blockly.user.keyMap.map_);
-  for (var i = 0; i < keys.length; i++) {
-    var key = keys[i];
-    var action = Blockly.user.keyMap.map_[key];
-    if (actionName === action.name) {
+  for (var i = 0, key; key = keys[i]; i++) {
+    if (Blockly.user.keyMap.map_[key] === action) {
```

Comment:
This won't work with the actions returned by getKeyMap currently since it's making copies of the action objects. For checking equality I think you do want to compare the names.